src/java/org/apache/cassandra/index/sai/disk/io/RAMIndexOutput.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.io;
+
+import java.io.IOException;
+
+import org.apache.lucene.store.GrowableByteArrayDataOutput;
+import org.apache.lucene.store.IndexOutput;
+
+// Use in place of RAMOutputStream (which has monitor locks)
```

Comment:
`RAMOutputStream` doesn't exist

Reply:
It's in lucene-core I think

Reply:
In any case, I'd make this a proper class JavaDoc. Also perhaps throw a `@NotThreadSafe` annotation on it.

Reply:
Aside: Does `RAMOutputStream ` have monitor locks? Anyway, it's deprecated in Lucene...

Reply:
> Aside: Does RAMOutputStream  have monitor locks? 

Yes, the `RAMFile` it uses for storage has a number of `synchronized` blocks in it. Either way, as you say, it's deprecated so shouldn't be being used anyway.